### PR TITLE
Add qjit compatibility note to molecular_hamiltonian

### PR DESCRIPTION
**Pull request by soranjh**
**Context:**
Add a note to the `molecular_hamiltonian` function saying that it is not currently compatible with `qjit`. [sc-80441]

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/qchem/hamiltonian.py [hunk 1]
```diff
@@ -309,6 +309,8 @@ def molecular_hamiltonian(*args, **kwargs):
         The ``molecular_hamiltonian`` function accepts a ``Molecule`` object as its first argument.
         Look at the `Usage Details` for more details on the old interface.
 
+        The ``molecular_hamiltonian`` function is not currently compatible with :func:`~.qjit` and ``jit``.
```
Comment (isaacdevlugt) [inline pennylane/qchem/hamiltonian.py:312]:
```suggestion
        The ``molecular_hamiltonian`` function is not currently compatible with :func:`~.qjit` and ``jax.jit``.
```

File: pennylane/qchem/molecule.py [hunk 2]
```diff
@@ -37,6 +37,9 @@ class Molecule:
 
     The molecule object can be passed to functions that perform a Hartree-Fock calculation.
 
+    .. note::
+        :class:`~.qchem.Molecule` is not currently compatible with :func:`~.qjit` and ``jit``.
```
Comment (isaacdevlugt) [inline pennylane/qchem/molecule.py:41]:
```suggestion
        :class:`~.qchem.Molecule` is not currently compatible with :func:`~.qjit` and ``jax.jit``.
```

**Review by isaacdevlugt — APPROVED** [review]
Looks good! Minor comment: the "Note:" section is after the parameters/returns section in `molecular_hamiltonian` and before the parameters/returns section in `Molecule`

**Comment by isaacdevlugt** [comment]
Oh! Don't forget a changelog entry :)

**Review by KetpuntoG — APPROVED** [review]

**Pull request merged**